tests/utils/utils.go

```diff
@@ -0,0 +1,333 @@
+func WaitForAllValidatorsToAcceptBlock(ctx context.Context, nodeURIs []string, blockchainID ids.ID, height uint64) {
+	for i, uri := range nodeURIs {
+		chainAWSURI := HttpToWebsocketURI(uri, blockchainID.String())
+		log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI)
+		client, err := ethclient.Dial(chainAWSURI)
+		Expect(err).Should(BeNil())
+
+		// Loop until each node has advanced to >= the height of the block that emitted the warp log
+		for {
+			block, err := client.BlockByNumber(ctx, nil)
+			Expect(err).Should(BeNil())
+			if block.NumberU64() >= height {
+				log.Info("client accepted the block containing SendWarpMessage", "client", i, "height", block.NumberU64())
+				break
+			}
+		}
+	}
+}
+
+// Constructs and sends a transaction containing a warp message for the destination chain.
+// Returns the signed transaction.
+func ConstructAndSendWarpTransaction(
+	ctx context.Context,
+	warpMessageBytes []byte,
+	requiredGasLimit *big.Int,
+	teleporterContractAddress common.Address,
+	fundedAddress common.Address,
+	fundedKey *ecdsa.PrivateKey,
+	wsClient ethclient.Client,
+	chainID *big.Int,
+) (*types.Transaction, *types.Receipt) {
+	// Construct the transaction to send the Warp message to the destination chain
+	log.Info("Constructing transaction for the destination chain")
+	signedMessage, err := avalancheWarp.ParseMessage(warpMessageBytes)
+	Expect(err).Should(BeNil())
+
+	numSigners, err := signedMessage.Signature.NumSigners()
+	Expect(err).Should(BeNil())
+
+	gasLimit, err := teleporter.CalculateReceiveMessageGasLimit(numSigners, requiredGasLimit)
+	Expect(err).Should(BeNil())
+
+	callData, err := teleporter.PackReceiveCrossChainMessage(teleporter.ReceiveCrossChainMessageInput{
+		MessageIndex:         0,
+		RelayerRewardAddress: fundedAddress,
+	})
+	Expect(err).Should(BeNil())
+
+	baseFee, err := wsClient.EstimateBaseFee(ctx)
+	Expect(err).Should(BeNil())
+
+	gasTipCap, err := wsClient.SuggestGasTipCap(ctx)
+	Expect(err).Should(BeNil())
+
+	nonce, err := wsClient.NonceAt(ctx, fundedAddress, nil)
+	Expect(err).Should(BeNil())
+
+	gasFeeCap := baseFee.Mul(baseFee, big.NewInt(2))
+	gasFeeCap.Add(gasFeeCap, big.NewInt(2500000000))
+	destinationTx := predicateutils.NewPredicateTx(
+		chainID,
+		nonce,
+		&teleporterContractAddress,
+		gasLimit,
+		gasFeeCap,
+		gasTipCap,
+		big.NewInt(0),
+		callData,
+		types.AccessList{},
+		warp.ContractAddress,
+		signedMessage.Bytes(),
+	)
+
+	// Sign and send the transaction on the destination chain
+	signer := types.LatestSignerForChainID(chainID)
+	signedTx, err := types.SignTx(destinationTx, signer, fundedKey)
+	Expect(err).Should(BeNil())
+
+	log.Info("Sending transaction to destination chain")
+	_, receipt := SendTransactionAndWaitForAcceptance(ctx, wsClient, signedTx)
+
+	return signedTx, receipt
+}
+
+// Constructs the aggregate signature, packs the Teleporter message, and relays to the destination
+func RelayMessage(
+	ctx context.Context,
+	block *types.Header,
```

Comment:
Let's just take the blockhash as a parameter here, we don't need anything else from the header.